### ISSUE TITLE
Change CHANGELOG format

### ISSUE THREAD

**Issue by ybiquitous** [stylelint/stylelint#3901]
> Clearly describe the documentation issue

Current [CHANGELOG.md](https://github.com/stylelint/stylelint/blob/7598f59be2f8ea0a956c040746b959a9d80456bf/CHANGELOG.md) has some `h1` level headings.

I think this is invalid as an HTML document (see https://stylelint.io/CHANGELOG/).

> What solution would you like to see?

For example, I suggest rewriting it like this ⬇️ 

```markdown
# Change Log

All notable changes to this project will be documented in this file. See also [the release page](https://github.com/stylelint/stylelint/releases).

## 9.11.0

- Item 1
- Item 2
- ...

## 9.10.0

- Item 1
- Item 2
- ...
```

**Comment by jeddy3** [comment]
> For example, I suggest rewriting it like this ⬇️

It's a good idea. However, I think we have the current format because the [tool we use to automatically generate the GitHub release from the CHANGELOG](https://github.com/MoOx/github-release-from-changelog#github-release-from-changelog) requires it.

We can certainly investigate other tools and approaches, though. Although, this is not something I'll personally have time to look into.

**Comment by jeddy3** [comment]
> What do you think about it?

Looks good. We always link to the changelog in each release tweet, and it bothers me that the title of the linked page is usually one version behind e.g. for release `9.10.0` the title of the link is `9.9.0` 
https://twitter.com/stylelint/status/1085485930348273665, having it say "Changelog" would be better.

Let's go with a hybrid approach as I don't think it's viable to adopt the "keep a changelog" _entirely_ at this late stage i.e. adopting the "Fixed", "Added", "Changed" subheadings.

Let's use:

```markdown
# Changelog

All notable changes to this project are documented in this file.

## [Head]

- Added: ...
- Fixed: ...
- Fixed: ...
...

## [9.10.0]

- Fixed: ...
- Fixed: ...
...
```

I believe that'll work and will just require a PR to updated the changelog to:

1. change the versions numbers to 2nd level headings
2. wrap those versions numbers in square brackets

And update the [release docs](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/releases.md#process) to change "Replace `# Head` in CHANGELOG.md with this new version number e.g. `# 8.1.2`" to "Replace `## [Head]` in CHANGELOG.md with this new version number e.g. `## [8.1.2]`".

**Comment by ybiquitous** [comment]
I completely agree with your idea! 🙆‍♂️
